### PR TITLE
Speed-up `NodeRegistrationTest`

### PR DESCRIPTION
**Pull request by vkolomeyko**
There was a problem identified with `NetworkMapServer` where response cache interval has not been correctly set for it.
This resulted delay of 1 minute in all the nodes discovering each other on the NetworkMap.

After the changes the test running time is down to 55 seconds from around 2 minutes previously.

### CODE REVIEW

File: node/src/integration-test/kotlin/net/corda/node/utilities/registration/NodeRegistrationTest.kt [hunk 1]
```diff
@@ -49,6 +47,7 @@ class NodeRegistrationTest {
         private val notaryName = CordaX500Name("NotaryService", "Zurich", "CH")
         private val aliceName = CordaX500Name("Alice", "London", "GB")
         private val genevieveName = CordaX500Name("Genevieve", "London", "GB")
+        private val log = loggerFor<NodeRegistrationTest>()
```
Comment (shamsasari) [inline node/src/integration-test/kotlin/net/corda/node/utilities/registration/NodeRegistrationTest.kt]: `contextLogger()`

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt [hunk 2]
```diff
@@ -32,13 +34,14 @@ import javax.ws.rs.core.Response
 import javax.ws.rs.core.Response.ok
 import javax.ws.rs.core.Response.status
 
-class NetworkMapServer(private val cacheTimeout: Duration,
+class NetworkMapServer(private val responseValidityDuration: Duration,
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt]: `pollInterval` is more accurate I think

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt [hunk 3]
```diff
@@ -108,36 +111,38 @@ class NetworkMapServer(private val cacheTimeout: Duration,
...
-                nodeInfoMap[signedNodeInfo.raw.hash] = signedNodeInfo
-                ok()
-            } catch (e: Exception) {
-                when (e) {
-                    is SignatureException -> status(Response.Status.FORBIDDEN).entity(e.message)
-                    else -> status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.message)
-                }
-            }.build()
-        }
+        fun publishNodeInfo(input: InputStream): Response =
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt]: Can you make this method have a code block. It makes the code less clear when you have a multi-line method that's an expression.

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt [hunk 4]
```diff
@@ -108,36 +111,38 @@ class NetworkMapServer(private val cacheTimeout: Duration,
...
         @POST
         @Path("ack-parameters")
         @Consumes(MediaType.APPLICATION_OCTET_STREAM)
-        fun ackNetworkParameters(input: InputStream): Response {
-            val signedParametersHash = input.readBytes().deserialize<SignedData<SecureHash>>()
-            val hash = signedParametersHash.verified()
-            latestAcceptedParametersMap[signedParametersHash.sig.by] = hash
-            return ok().build()
-        }
+        fun ackNetworkParameters(input: InputStream): Response =
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt]: ditto

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt [hunk 5]
```diff
@@ -170,11 +175,13 @@ class NetworkMapServer(private val cacheTimeout: Duration,
...
             }
             requireNotNull(requestedParameters)
-            return Response.ok(requestedParameters!!.serialize().bytes).build()
+            Response.ok(requestedParameters!!.serialize().bytes).build()
         }
 
         @GET
         @Path("my-hostname")
-        fun getHostName(): Response = Response.ok(myHostNameValue).build()
+        fun getHostName(): Response = logElapsedTime("My hostname") {
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt]: I doubt logging this has much benefit.

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt (hunk 2 above)
Comment (vkolomeyko) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt]: sure - will change

File: node/src/integration-test/kotlin/net/corda/node/utilities/registration/NodeRegistrationTest.kt (hunk 1 above)
Comment (vkolomeyko) [reply node/src/integration-test/kotlin/net/corda/node/utilities/registration/NodeRegistrationTest.kt]: Will change.

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt (hunk 3 above)
Comment (vkolomeyko) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt]: I have largely undone my changes around performance logging to `NetworkMapServer` - it is indeed adding little additional insight.

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt (hunk 5 above)
Comment (vkolomeyko) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/network/NetworkMapServer.kt]: Agree.

**Pull request merged**